variants/arduino_101/linker_scripts/flash.ld

```diff
@@ -40,8 +40,8 @@ OUTPUT_FORMAT("elf32-littlearc", "elf32-bigarc", "elf32-littlearc")
 MEMORY
     {
     FLASH                 (rx) : ORIGIN = 0x40034000, LENGTH = 152K
-    SRAM                  (wx) : ORIGIN = 0xa800e000, LENGTH = 24K
```

Comment:
I thought "rwx" and "wx" were the same thing (i.e. "w" signifies both read and write).  I'm not 100% sure though. Can somebody explain why this is needed?